stages: allow bootc.install-to-filesystem work without selinux (HMS-3453)

mvo5 opened the pull request:
By default "bootc" will refuse to work on a non-selinux system if the bootc container requires selinux. This is a sensible approach in general but for us it's tricky because we want to be able to generate images when running on developer machines or CI machines that may not necessarily have selinux. So make bootc more relaxed.

Fwiw with the recent improvements in bootc this seems to be working fine for me (i.e. installing from non-selinux).

Note that once https://github.com/containers/bootc/pull/420 is available in our various downstreams this can be dropped.

supakeen commented on stages/org.osbuild.bootc.install-to-filesystem:
```diff
@@ -12,6 +13,19 @@ def main(options, inputs, paths):
+    # in general but for us it's tricky because we want to be able to
+    # generate images when running on developer machines or CI machines
+    # that may not necessarily have selinux. So make bootc more relaxed.
+    env["BOOTC_SKIP_SELINUX_HOST_CHECK"] = "true"
```
I don't like doing this unconditionally; can we tell it through the schema?

mvo5 replied:
Yeah, it's not great, I agree. I was thinking about this too (assuming you mean a schema value like "skip-selinux-host-check: bool" that then would be put into the manifest):
This seems not an ideal place because it feels a bit odd that the manifest.json would be dependent on what host osbuild runs on. Also the schema could be generated on a different machine, it could be generated a long time ago on a host with selinux and then it got disabled and suddently the manifest stops works.

To make it conditional we would have to find a way to pass it from osbuild to the stage without a schema/manifest (IMO), I don't recall we have a mechanism today that can do that?

But yeah, if you/Colin feels strongly about this we need to make it conditional.

cgwalters replied:
> but for us it's tricky because we want to be able to generate images when running on developer machines or CI machines that may not necessarily have selinux.

Handling it for CI is trivial, set the environment variable in the CI flow which is exactly what bootc is doing for its CI.

I wouldn't really distinguish "developer" from "user" in general here - having it on by default here  I think this tantamount to supporting it in practice.  So maybe we should just do that.

If it works for you now, then yeah we can just go ahead and change bootc and claim to support it and just try to keep testing it.

I know for sure we need to support it longer term so it may be the right time to just say we are, and if it breaks we fix it.

supakeen replied:
Does this affect the output image?

cgwalters replied:
No effect on output.

cgwalters commented:
> In addition this commit also enables debug logging from bootc. The rational is that at least for now it will yield valuable information about what is going on and arguable we are already spaming the terminal too much

cgwalters commented:
> In addition this commit also enables debug logging from bootc. The rational is that at least for now it will yield valuable information about what is going on and arguable we are already spaming the terminal too much

Hmm.  There's quite a lot of logging at that level.  Now what I think we definitely should do is make it trivial/easy to turn *on* the logging when needed; I did just that in e.g. https://gitlab.com/bootc-org/podman-bootc-cli/-/merge_requests/42/diffs?commit_id=6cd5a20c1a7173e63f943a4bb314c5be468f140d but some sort of generalized mechanism like `bootc-image-builder --stage-env=stagename:envvar:value` like `bootc-image-builder --stage-env=org.osbuild.bootc-install:RUST_LOG=debug` ?

cgwalters commented:
I did https://github.com/containers/bootc/pull/420 related to this

mvo5 commented:
> I did [containers/bootc#420](https://github.com/containers/bootc/pull/420) related to this

Thank you! That is great. Fwiw, I agree that we need to support this eventually anyway so :+1: for doing it now. Once the other one is merged I will close this one and think a bit more about the debug switch. My gut feeling is slightly against passing unrestricted envs into stages as it my give a bit too much freedom to make the builds less repeatable in a stable fashion but at this level the user can do a lot anyhow so maybe I'm overthinking it :)

ondrejbudai commented:
I'm a bit afraid that it will take some time for containers/bootc#420 to flow into all downstreams (not blaming anyone, this is just how things sometimes are). Since the bootc PR basically implements `BOOTC_SKIP_SELINUX_HOST_CHECK` as the default (I know, there are some differences), can we just proceed with merging this PR to unblock us? The logging change seems to be controversial, so let's drop it/split it?

mvo5 commented:
> I'm a bit afraid that it will take some time for [containers/bootc#420](https://github.com/containers/bootc/pull/420) to flow into all downstreams (not blaming anyone, this is just how things sometimes are). Since the bootc PR basically implements `BOOTC_SKIP_SELINUX_HOST_CHECK` as the default (I know, there are some differences), can we just proceed with merging this PR to unblock us? The logging change seems to be controversial, so let's drop it/split it?

Sure thing! I updated the PR and dropped the logging change for now. We can revisit that again in the future.

ondrejbudai commented:
Can you also adjust the unit test?

mvo5 commented:
> Can you also adjust the unit test?

Thanks! Silly me, fixed.

ondrejbudai reviewed (approved):
Let's merge this! :)
